Sources/StreamChat/Repositories/SyncRepository.swift

```diff
@@ -108,10 +137,76 @@ class SyncRepository {
                 }
                 return
             }
-
-            self?.syncLocalState(lastSyncAt: lastSyncAt, completion: completion)
+            if self?.usesV2Sync == true {
+                self?.syncLocalStateV2(lastSyncAt: lastSyncAt, completion: completion)
+            } else {
+                self?.syncLocalState(lastSyncAt: lastSyncAt, completion: completion)
+            }
         }
     }
+    
+    // MARK: - V2
+    
+    /// Runs offline tasks and updates the local state for channels
+    ///
+    /// Recovery mode (pauses regular API requests while it is running)
+    /// 1. Enter recovery
+    /// 2. Runs offline API requests
+    /// 3. Exit recovery
+    ///
+    /// Background mode (other regular API requests are allowed to run at the same time)
+    /// 1. Collect all the **active** channel ids (from instances of `Chat`, `ChannelList`, `ChatChannelController`, `ChatChannelListController`)
+    /// 2. Apply updates from the /sync endpoint for these channels
+    /// 3. Refresh channel lists (channels for current pages in `ChannelList`, `ChatChannelListController`)
+    private func syncLocalStateV2(lastSyncAt: Date, completion: @escaping () -> Void) {
+        let context = SyncContext(lastSyncAt: lastSyncAt)
+        var operations: [Operation] = []
+        let start = CFAbsoluteTimeGetCurrent()
+        log.info("Starting to refresh offline state", subsystems: .offlineSupport)
+        
+        //
+        // Recovery mode operations (other API requests are paused)
+        //
+        if config.isLocalStorageEnabled {
+            apiClient.enterRecoveryMode()
+            operations.append(ExecutePendingOfflineActions(offlineRequestsRepository: offlineRequestsRepository))
```

Comment:
I assume there are no changes in this logic, right? We have some logic for sending new messages - after some interval they shouldn't be sent when you come online. Just making sure we still have that.